src/routes/login/+page.svelte

```diff
@@ -4,22 +4,62 @@
 	import image3x from '$lib/assets/images/image@3x.webp';
 	import { Button } from '$lib/components/ui/button';
 	import { Input } from '$lib/components/ui/input';
-	import { Label } from '$lib/components/ui/label';
+	import { defaults, superForm } from 'sveltekit-superforms';
+	import { loginSchema } from '../../schema/login';
+	import { zod } from 'sveltekit-superforms/adapters';
+	import serviceFactory from '$lib/services/serviceFactory';
+	import { FormControl, FormField, FormFieldErrors, FormLabel } from '$lib/components/ui/form';
+
+	const form = superForm(defaults(zod(loginSchema)), {
+		SPA: true,
+		validators: zod(loginSchema),
+		onUpdate: async ({ form }) => {
+			if (form.valid) {
+				const { email, password } = form.data;
+				const userService = serviceFactory.get('UserService');
+				await userService.loginUser(email, password);
+			}
+		}
+	});
+	const { form: formData, errors, enhance } = form;
 </script>
 
-<div class="min-w-screen min-h-screen flex justify-center items-center">
-	<div class="hidden desktop:flex flex-col items-center justify-center w-[50vw]">
-		<img src={image} srcset="{image} 720w, {image2x} 1440w, {image3x} 2160w" sizes="(min-width: 1440px) 50vw," alt="Background" />
+<div class="min-w-screen flex min-h-screen items-center justify-center">
+	<div class="hidden w-[50vw] flex-col items-center justify-center desktop:flex">
+		<img
+			src={image}
+			srcset="{image} 720w, {image2x} 1440w, {image3x} 2160w"
+			sizes="(min-width: 1440px) 50vw,"
+			alt="Background"
+		/>
 	</div>
-	<div class="desktop:w-[50vw] flex flex-grow flex-col items-center justify-center">
-		<div class="w-mobile p-6">
-			<h1 class="text-center text-2xl font-semibold font-inter text-card-foreground mb-2">Login</h1>
-			<p class="text-center mb-6 text-muted-foreground">Enter your email below<br/>to login to your account</p>
-			<form class="flex flex-col justify-center" method="POST">
-                <Label class="mb-1.5" for="email">Email</Label>
-				<Input class="mb-4" type="email" id="email" name="email" placeholder="name@example.com" />
-                <Label class="mb-1.5" for="password">Password</Label>
-                <Input class="mb-6" type="password" id="password" name="password" />
+	<div class="flex flex-grow flex-col items-center justify-center desktop:w-[50vw]">
+		<div class="w-mobile p-6 desktop:p-4">
+			<h1 class="mb-2 text-center font-inter text-2xl font-semibold text-card-foreground">Login</h1>
+			<p class="mb-6 text-center font-inter text-sm font-normal text-muted-foreground">
+				Enter your email below<br />to login to your account.
+			</p>
+			<form class="flex flex-col justify-center" method="POST" use:enhance>
+				<FormField {form} name="email">
+					<FormControl let:attrs>
+						<FormLabel class="block mb-1.5">Email</FormLabel>
+						<Input
+							{...attrs}
+							class={$errors.email ? "mb-1.5" : "mb-4"}
```

Comment:
Good spot to use `cn()` inside your class. An example would look like this: `class={cn("mb-4", {"mb-1.5": $errors.email})`

If $errors.email is true, the class mb-1.5 will be applied, overriding the mb-4 class. This is because mb-1.5 will conditionally be added when $errors.email is true, and the last margin class takes precedence.

Reply:
That makes sense. It doesn't seem like very useful in this case... is it more of a style guide thing? I could understand how if there were 2-3 conditional statement. It could prove useful.

Reply:
Personally, I think cn(...) comes off cleaner looking. It allows you to see what the default CSS styles are—on the left side—while clearly showing that there's some logic being applied on the right side.

To break it down more, using cn(...) helps organize your styles. It simplifies conditional logic for applying classes based on some state or condition. So, for example, if you had more complex logic—say $errors.email, $errors.password, and $errors.username all needing different margin classes—using this pattern helps keep everything in one place without repeating yourself or cluttering your code with multiple if-else blocks or ternary operators inside your class attributes.

It's also great for ensuring you don’t accidentally overwrite default styles. Like in this case, mb-4 is always applied unless $errors.email is true, in which case mb-1.5 takes over. That way, you have a clear fallback style and a straightforward way to adjust the UI based on specific conditions.

In bigger projects, this pattern tends to improve readability and maintainability, especially when you have multiple UI states that require different styling.

Reply:
That makes a lot of sense. I think the syntax does indicate a fallback style. And it does seem more clean if you think of it that way. It is a helpful pattern to know. Thanks for the comment!